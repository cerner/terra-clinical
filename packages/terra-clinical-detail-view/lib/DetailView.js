'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

require('terra-base/lib/baseStyles');

require('./DetailView.scss');

var _DetailList = require('./DetailList');

var _DetailList2 = _interopRequireDefault(_DetailList);

var _DetailListItem = require('./DetailListItem');

var _DetailListItem2 = _interopRequireDefault(_DetailListItem);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  /**
   * The title to display
   **/
  title: _propTypes2.default.string,

  /**
   * List of subtitle strings
   **/
  subtitles: _propTypes2.default.arrayOf(_propTypes2.default.string),

  /**
   * Display for visualization data
   **/
  graph: _propTypes2.default.element,

  /**
   * Array of elements to display detail information
   **/
  details: _propTypes2.default.arrayOf(_propTypes2.default.element),

  /**
   * Text to be displayed at the footer of the view
   **/
  footer: _propTypes2.default.string,

  /**
   * Indicates if sections should be devided
   **/
  isDivided: _propTypes2.default.bool
};

var defaultProps = {
  title: '',
  subtitles: [],
  details: [],
  isDivided: true
};

var DetailView = function DetailView(_ref) {
  var title = _ref.title,
      subtitles = _ref.subtitles,
      graph = _ref.graph,
      details = _ref.details,
      footer = _ref.footer,
      isDivided = _ref.isDivided,
      customProps = _objectWithoutProperties(_ref, ['title', 'subtitles', 'graph', 'details', 'footer', 'isDivided']);

  var attributes = _extends({}, customProps);
  attributes.className = (0, _classnames2.default)(['terraClinical-DetailView', attributes.className]);

  var divider = null;
  var dividedDetails = [];

  var subtitleElements = subtitles.map(function (subtitle, i) {
    return (
      // eslint-disable-next-line react/no-array-index-key
      _react2.default.createElement(
        'div',
        { className: 'terraClinical-DetailView-subtitle', key: i },
        subtitle
      )
    );
  });

  if (isDivided) {
    divider = _react2.default.createElement('hr', { className: 'terraClinical-DetailView-divider' });

    for (var i = 0; i < details.length; i += 1) {
      dividedDetails.push(details[i]);
      dividedDetails.push(divider);
    }
  } else {
    dividedDetails = details;
  }

  return _react2.default.createElement(
    'div',
    attributes,
    _react2.default.createElement(
      'div',
      { className: 'terraClinical-DetailView-title' },
      _react2.default.createElement(
        'h1',
        { className: 'terraClinical-DetailView-primaryText' },
        title
      ),
      subtitleElements
    ),
    divider,
    graph,
    divider,
    dividedDetails,
    _react2.default.createElement(
      'div',
      { className: 'terraClinical-DetailView-footerText' },
      footer
    )
  );
};

DetailView.propTypes = propTypes;

DetailView.defaultProps = defaultProps;
DetailView.DetailList = _DetailList2.default;
DetailView.DetailListItem = _DetailListItem2.default;

exports.default = DetailView;