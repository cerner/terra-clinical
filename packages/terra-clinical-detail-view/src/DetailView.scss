:local {
  .detail-view {
    padding: 1.428rem;

    > * {
      margin-bottom: 1.428rem;
    }
  }

  .title {
    line-height: 1.428;
    margin-bottom: 1.785rem;
  }

  .primary-text {
    font-size: 1.428rem;
    margin-bottom: 0;
    margin-top: 0;
  }

  .footer-text {
    color: #909496;
    font-size: 1rem;
    line-height: 1.428;
    text-align: center;
  }

  .divider {
    border-color: #d3d4d5;
    border-style: solid;
    border-width: 1px 0 0;
    margin: 0;
    padding-bottom: 0.714rem;
  }

  .subtitle {
    color: #bcbfc0;
    font-size: 1.286rem;

    &:last-child {
      margin: 0;
    }
  }
}
