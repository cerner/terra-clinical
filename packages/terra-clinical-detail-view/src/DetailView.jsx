import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import styles from './DetailView.scss';
import DetailList from './DetailList';
import DetailListItem from './DetailListItem';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The title to display
   */
  title: PropTypes.string,

  /**
   * List of subtitle strings
   */
  subtitles: PropTypes.arrayOf(PropTypes.string),

  /**
   * Display for visualization data
   */
  graph: PropTypes.element,

  /**
   * Array of elements to display detail information
   */
  details: PropTypes.arrayOf(PropTypes.element),

  /**
   * Text to be displayed at the footer of the view
   */
  footer: PropTypes.string,

  /**
   * Indicates if sections should be devided
   */
  isDivided: PropTypes.bool,
};

const defaultProps = {
  title: '',
  subtitles: [],
  details: [],
  isDivided: true,
};

const DetailView = ({ title, subtitles, graph, details, footer, isDivided, ...customProps }) => {
  const attributes = Object.assign({}, customProps);
  attributes.className = cx(['detail-view',
    attributes.className,
  ]);

  let divider = null;
  let dividedDetails = [];

  const subtitleElements = subtitles.map((subtitle, i) => (
    // eslint-disable-next-line react/no-array-index-key
    <div className={cx('subtitle')} key={i}>{subtitle}</div>
  ));

  if (isDivided) {
    divider = (<hr className={cx('divider')} />);

    for (let i = 0; i < details.length; i += 1) {
      dividedDetails.push(details[i]);
      dividedDetails.push(divider);
    }
  } else {
    dividedDetails = details;
  }

  return (
    <div {...attributes}>
      <div className={cx('title')}>
        <h1 className={cx('primary-text')}>{title}</h1>
        {subtitleElements}
      </div>
      {divider}
      {graph}
      {divider}
      {dividedDetails}
      <div className={cx('footer-text')}>
        {footer}
      </div>
    </div>
  );
};

DetailView.propTypes = propTypes;

DetailView.defaultProps = defaultProps;
DetailView.DetailList = DetailList;
DetailView.DetailListItem = DetailListItem;

export default DetailView;
