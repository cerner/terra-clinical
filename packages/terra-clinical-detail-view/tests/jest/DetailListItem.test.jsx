import React from 'react';
import DetailView from '../../src/DetailView';

const defaultVariety = <DetailView.DetailListItem item={<p>Test</p>} />;

// Snapshot Tests
it('should render a default component', () => {
  const wrapper = render(defaultVariety);
  expect(wrapper).toMatchSnapshot();
});

// Structure Tests
it('should have the class detail-list-item', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper.prop('className')).toContain('detail-list-item');
});
