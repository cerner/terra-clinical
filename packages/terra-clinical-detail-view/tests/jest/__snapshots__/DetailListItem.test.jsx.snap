exports[`test should render a default component 1`] = `
<div
  class="detail-list-item">
  <p>
    Test
  </p>
</div>
`;
